src/main.ts

```diff
@@ -785,6 +785,8 @@ export default class ObsidianGit extends Plugin {
         amend?: boolean;
     }): Promise<boolean> {
         if (!(await this.isAllInitialized())) return false;
+        if (!(this.settings.globalEnable)) return false; // if globalEnable-checkbox is disabled, return without committing
```

Comment:
I think you should still be able to manually commit via Obsidian. Only the automatic commits should be disabled, don't they? So I would not add that check here, but reset the automatics and don't start them.